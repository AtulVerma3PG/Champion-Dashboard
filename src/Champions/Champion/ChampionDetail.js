import React from "react";
import PropTypes from "prop-types";
import * as ReactBootStrap from "react-bootstrap";

const ChampionDetail = (props) => {
  const { location } = props;
  const { champion } = location.state;
  return (
    <div className="container">
      <h2>Champion Detail</h2>
      <br />
      <br />
      <br />

      <ReactBootStrap.Table responsive="lg" striped bordered hover>
        <thead>
          <th colSpan="2">
            <img src={champion.image_url} alt={champion.image_url} />
          </th>
        </thead>
        <tbody>
          <tr>
            <td>id </td>
            <td>
              {" "}
              {champion.id}
              {" "}
            </td>
          </tr>
          <tr>
            <td>name </td>
            <td>
              {" "}
              {champion.name}
              {" "}
            </td>
          </tr>
          <tr>
            <td>armor </td>
            <td>
              {" "}
              {champion.armor}
              {" "}
            </td>
          </tr>
          <tr>
            <td>armorperlevel </td>
            <td>
              {" "}
              {champion.armorperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackdamage </td>
            <td>
              {" "}
              {champion.attackdamage}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackdamageperlevel </td>
            <td>
              {" "}
              {champion.attackdamageperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackrange </td>
            <td>
              {" "}
              {champion.attackrange}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackspeedoffset </td>
            <td>
              {" "}
              {champion.attackspeedoffset}
              {" "}
            </td>
          </tr>
          <tr>
            <td>attackspeedperlevel </td>
            <td>
              {" "}
              {champion.attackspeedperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>crit </td>
            <td>
              {" "}
              {champion.crit}
              {" "}
            </td>
          </tr>
          <tr>
            <td>critperlevel </td>
            <td>
              {" "}
              {champion.critperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hp </td>
            <td>
              {" "}
              {champion.hp}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpperlevel </td>
            <td>
              {" "}
              {champion.hpperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpregen </td>
            <td>
              {" "}
              {champion.hpregen}
              {" "}
            </td>
          </tr>
          <tr>
            <td>hpregenperlevel </td>
            <td>
              {" "}
              {champion.hpregenperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>movespeed </td>
            <td>
              {" "}
              {champion.movespeed}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mp </td>
            <td>
              {" "}
              {champion.mp}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpperlevel </td>
            <td>
              {" "}
              {champion.mpperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpregen </td>
            <td>
              {" "}
              {champion.mpregen}
              {" "}
            </td>
          </tr>
          <tr>
            <td>mpregenperlevel </td>
            <td>
              {" "}
              {champion.mpregenperlevel}
              {" "}
            </td>
          </tr>

          <tr>
            <td>spellblock </td>
            <td>
              {" "}
              {champion.spellblock}
              {" "}
            </td>
          </tr>
          <tr>
            <td>spellblockperlevel </td>
            <td>
              {" "}
              {champion.spellblockperlevel}
              {" "}
            </td>
          </tr>
          <tr>
            <td>videogame_versions</td>
            <td>
              {champion.videogame_versions.map((version) => `${version} ,`)}
            </td>
          </tr>
        </tbody>
      </ReactBootStrap.Table>
    </div>
  );
};

ChampionDetail.propTypes = {
  location: PropTypes.object.isRequired,
};

export default ChampionDetail;
