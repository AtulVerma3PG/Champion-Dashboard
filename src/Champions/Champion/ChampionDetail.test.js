/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, mount } from "enzyme";
import ChampionDetail from "./ChampionDetail";

configure({ adapter: new Adapter() });
const location = {
  hash: "",
  key: "7kdb4z",
  pathname: "/SubmitTest",
  search: "",
  state: {
    articles: [

    ],
    searches: [

    ],
    watchlist: [

    ],
    searchedText: "",
    champion: [
      { videogame_versions: [10] },
    ],
    sortBy: "Asc",
  },
};
test("App Snapshot test", () => {
  const component = renderer.create(
    <ChampionDetail location={location} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const history = createMemoryHistory();
history.push("/");
describe("Test Button component", () => {
  it("Test click event", () => {
    const historyMock = { push: jest.fn() };
    const shallowCopy = shallow(<ChampionDetail history={historyMock} location={location} />);
    shallowCopy.find("button").simulate("onClick");
    expect(history.location.pathname).toEqual("/");
  });
});
