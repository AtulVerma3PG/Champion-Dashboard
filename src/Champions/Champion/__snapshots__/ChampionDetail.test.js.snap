// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot test 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      My Watchlist
    </span>
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Home
    </button>
  </nav>
  <div
    className="table-responsive-lg"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <th
          colSpan="2"
        >
          <img />
        </th>
      </thead>
      <tbody>
        <tr>
          <td>
            id 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            name 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            armor 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            armorperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            attackdamage 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            attackdamageperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            attackrange 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            attackspeedoffset 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            attackspeedperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            crit 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            critperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            hp 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            hpperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            hpregen 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            hpregenperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            movespeed 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            mp 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            mpperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            mpregen 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            mpregenperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            spellblock 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            spellblockperlevel 
          </td>
          <td>
             
             
          </td>
        </tr>
        <tr>
          <td>
            videogame_versions
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
